TFW-2941 - Move US SearchBar into Header

jasonhale opened the pull request:
Moves the HeaderPane out of the base layout, updates some styles accordingly, and adds the SearchBar into the Header within Universal search only.

Also added some page layout styles from the new design pattern for padding on pages.

This work is required for TFW-2941 on the RCI repo.

Will clean up the commits after PR is approved and ready to go.

troyblank commented on assets/sass/base/_mixins.scss:
```diff
@@ -50,3 +50,24 @@
+    @include breakpoint($breakpoint-size) {
+        @include page-gutters('small');
+    }
+}
```
A couple issues I noticed with this; The header seems to have too much spacing at the bottom compared to the design. Also the rocket in mobile view is now off the bottom of the page where before it sat at the bottom.

![screen shot 2019-03-05 at 10 10 09 am](https://user-images.githubusercontent.com/996725/53815221-411dde00-3f2f-11e9-88a7-fa1c95d7d4c5.png)

![screen shot 2019-03-05 at 10 11 30 am](https://user-images.githubusercontent.com/996725/53815229-43803800-3f2f-11e9-8d82-b9d7cc382e25.png)

jasonhale commented:
@troyblank - pulled in latest from `develop` and updated a few styles.  Also modified the "clear" button to match the design docs.  Talked to Casey, it's a one-off style.

![screen shot 2019-03-05 at 11 54 19 am](https://user-images.githubusercontent.com/3496073/53826471-49314a00-3f3e-11e9-92df-b20e6d801df3.png)
